### PR TITLE
Add fixtures for question model testing.

### PR DESCRIPTION
**Pull request by kasya**
Divide tests into smaller test cases for each particular check.

### CODE REVIEW

File: tests/question/test_models.py [hunk 1]
```diff
@@ -10,8 +10,8 @@ class QuestionModelTestCase(TestCase):
   """Test cases for Question models."""
 
   fixtures = [
-      'exam_attempt.json', 'user.json', 'exam.json', 'subject.json',
-      'question.json', 'question_category.json', 'answer.json'
+      'answer.json', 'exam.json', 'exam_attempt.json', 'question.json',
```
Comment (arkid15r) [inline tests/question/test_models.py:13]: 👍

File: tests/question/test_views.py [hunk 2]
```diff
@@ -17,68 +17,65 @@ class QuestionViewTestCase(TestCase):
...
-      'question.json', 'question_category.json', 'answer.json'
+      'answer.json', 'exam.json', 'exam_attempt.json', 'question.json',
+      'question_category.json', 'subject.json', 'user.json'
   ]
   user_password = 'mypassword'
 
+  def setUp(self):
+
+    self.attempt = ExamAttempt.objects.get(id=1)
+    self.user = User.objects.get(id=4)
```
Comment (arkid15r) [inline tests/question/test_views.py]: Better ordering? :)

File: tests/question/test_views.py [hunk 3]
```diff
@@ -17,68 +17,65 @@ class QuestionViewTestCase(TestCase):
...
-          reverse('question_api', kwargs={'attempt_id': attempt.id})).render()
+          reverse('question_api',
+                  kwargs={'attempt_id': self.attempt.id})).render()
 
       data = json.loads(response.content)
       self.assertEqual(response.status_code, 200)
-      self.assertEqual(QuestionSerializer(question).data, data)
-      AnswerAttempt.objects.create(attempt=attempt, question=question)
+      self.assertEqual(QuestionSerializer(self.question).data, data)
+      AnswerAttempt.objects.create(attempt=self.attempt, question=self.question)
```
Comment (arkid15r) [inline tests/question/test_views.py]: What does this one do?

File: tests/question/test_views.py [hunk 4]
```diff
@@ -17,68 +17,65 @@ class QuestionViewTestCase(TestCase):
...
   def test_api_get_question_anonymous(self):
     """Try to access page with anonymous user."""
 
-    attempt = ExamAttempt.objects.get(id=1)
     response = self.client.get(
-        reverse('question_api', kwargs={'attempt_id': attempt.id}))
+        reverse('question_api', kwargs={'attempt_id': self.attempt.id}))
     self.assertEqual(response.status_code, 404)
 
   def test_api_get_question_wrong_user(self):
```
Comment (arkid15r) [inline tests/question/test_views.py]: What is right, what is wrong? :)
Unauthorized?

File: tests/question/test_views.py [hunk 5]
```diff
@@ -17,68 +17,65 @@ class QuestionViewTestCase(TestCase):
...
 
+    user = User.objects.get(id=5)
     self.client.login(username=user.username, password=self.user_password)
     response = self.client.get(
-        reverse('question_api', kwargs={'attempt_id': attempt.id}))
+        reverse('question_api', kwargs={'attempt_id': self.attempt.id}))
     self.assertEqual(response.status_code, 404)
 
   def test_api_get_question(self):
     """Check that method send correct data to frontend."""
```
Comment (arkid15r) [inline tests/question/test_views.py]: sends

File: tests/question/test_views.py [hunk 6]
```diff
@@ -89,33 +86,28 @@ def test_api_post_question_404(self):
     if required data is missing.
     """
 
-    attempt = ExamAttempt.objects.get(id=1)
-    user = User.objects.get(id=4)
-    self.client.login(username=user.username, password=self.user_password)
+    self.client.login(username=self.user.username, password=self.user_password)
     required_data = ({}, {'question_id': 1})
 
     for data in required_data:
```
Comment (arkid15r) [inline tests/question/test_views.py]: invalid_data?

File: tests/question/test_views.py [hunk 7]
```diff
@@ -89,33 +86,28 @@ def test_api_post_question_404(self):
...
-      response = self.client.post(reverse('question_api',
-                                          kwargs={'attempt_id': attempt.id}),
+      response = self.client.post(reverse(
+          'question_api', kwargs={'attempt_id': self.attempt.id}),
                                   data=data)
       self.assertEqual(response.status_code, 404)
 
   def test_api_post_question(self):
-    """Check that method creates Answer Attempt ans saves answer to db."""
+    """Check that method create Answer Attempt and save answer to db."""
```
Comment (arkid15r) [inline tests/question/test_views.py]: creates/saves

File: tests/question/test_views.py [hunk 8]
```diff
@@ -125,27 +117,24 @@ def test_api_post_question(self):
   def test_api_post_update_answers(self):
     """Check that answer id in db changes when we change our answers."""
 
-    question = Question.objects.get(id=1)
     correct_answer = Answer.objects.get(id=1)
     wrong_answer = Answer.objects.get(id=2)
-    user = User.objects.get(id=4)
-    exam = Exam.objects.get(id=1)
-    attempt = ExamAttempt.objects.create(user=user, exam=exam)
+    attempt = ExamAttempt.objects.create(user=self.user, exam=self.exam)
```
Comment (arkid15r) [inline tests/question/test_views.py]: Move before correct_answer

File: tests/question/test_views.py (hunk 3 above)
Comment (kasya) [reply tests/question/test_views.py]: > What does this one do?

Have no idea how it got there :)

**Review by arkid15r — APPROVED** [review]

**Pull request merged**